src/redux/books/BookSlice.js

```diff
@@ -1,36 +1,54 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+export const postData = createAsyncThunk('data/postData', async () => {
+  try {
+    const response = await axios.post('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/');
+    console.log(response.data);
+  } catch (error) {
+    console.log(error);
+  }
+});
+
+export const getListBook = createAsyncThunk('data/getListBook', async () => {
+  try {
+    const response = await axios.get('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/XISo38gwNhXdKdbu3B94/books');
+    console.log(response.data);
+    return response.data;
+  } catch (error) {
+    console.log(error);
+  }
+});
+
+export const postnewbook = createAsyncThunk('data/postnewbook', async (requestbook) => {
+  try {
+    const response = await axios.post('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/XISo38gwNhXdKdbu3B94/books', JSON.stringify(requestbook), {
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+    console.log(response);
+  } catch (error) {
+    console.log(error);
+  }
+});
+
+export const deletebook = createAsyncThunk('data/deletebook', async (IDD) => {
```

Comment:
- [ ] As for these 2 actions, when they are called, the state is not updated unless I manually refresh the page, I mean when a new book is added or a book is deleted, you need to manually reload the page for the changes to apply which is not a good idea at all, please fix it
![6](https://github.com/massazoum/Zoum_Bookstore/assets/77843050/514e8ccb-36f3-4776-9c84-4c486e04cc21)